### PR TITLE
SSL support for ES HTTP requests

### PR DESCRIPTION
**Pull request by gtschurwald**
If SSL and a cert-key pair are needed to hit ES endpoints, this will update collectd to work properly.

### CODE REVIEW

File: elasticsearch_collectd.py [hunk 1]
```diff
@@ -520,6 +526,15 @@ def configure_callback(conf):
             ENABLE_CLUSTER_STATS = bool(node.values[0])
         elif node.key == 'Interval':
             COLLECTION_INTERVAL = int(node.values[0])
+        elif node.key == 'HttpTlsEnabled':
+            if bool(node.values[0]):
+                ES_URL_SCHEME = "https"
```
Comment (natejlong) [inline elasticsearch_collectd.py:531]: I would have expected this to be treated like constants, from the case. I'd suggest you copy them to a local variable just for clarity

Comment (natejlong) [reply elasticsearch_collectd.py:531]: Alternatively, there might be a better pattern for using globals without implying constant that cleans this up. My python-fu is weak there though

File: elasticsearch_collectd.py [hunk 2]
```diff
@@ -628,10 +643,28 @@ def fetch_stats():
...
 
+# SSL requests wrapper
+class HTTPSClientAuthHandler(urllib2.HTTPSHandler):
+    def __init__(self, key_file, cert_file):
+        urllib2.HTTPSHandler.__init__(self)
+        self.key_file = key_file
+        self.cert_file = cert_file
+
+    def https_open(self, req):
+        return self.do_open(self.get_connection, req)
```
Comment (natejlong) [inline elasticsearch_collectd.py:654]: naive: Does this need to be closed at some point?

Comment (gtschurwald) [reply elasticsearch_collectd.py:654]: Looking at the base class this inherits from, it has a close method which doesn't do anything. 
```
    def close(self):
        # Only exists for backwards compatibility
        pass
```
So it doesn't seem like it's needed with this library. Good question though, normally I'd expect that to be needed.

File: elasticsearch_collectd.py (hunk 1 above)
Comment (gtschurwald) [reply elasticsearch_collectd.py:531]: Yep, I was primarily following the convention already in this file for these. I think there should be some way to indicate these variables should not be changed, another convention I've seen in python would be g_var_name to indicate a global that isn't a constant. I think adopting a constant-like naming scheme kind of makes sense here as these should only be set in the outer scope or in this configuration function.

File: elasticsearch_collectd.py (hunk 2 above)
Comment (natejlong) [reply elasticsearch_collectd.py:654]: Cool, thanks for looking into it

File: elasticsearch_collectd.py [hunk 3]
```diff
@@ -628,10 +643,28 @@ def fetch_stats():
...
+    def get_connection(self, host, timeout=300):
+        return httplib.HTTPSConnection(host, key_file=self.key_file, cert_file=self.cert_file, timeout=timeout)
+
+
 def fetch_url(url):
     response = None
     try:
-        response = urllib2.urlopen(url, timeout=10)
+        if ES_HTTP_TLS_ENABLED:
+            opener = urllib2.build_opener(HTTPSClientAuthHandler(ES_TLS_KEY_PATH, ES_TLS_CERT_PATH))
```
Comment (natejlong) [inline elasticsearch_collectd.py:664]: super minor nit: is the change in timeout here intentional/significant?

File: elasticsearch_collectd.py (hunk 1 above)
Comment (natejlong) [reply elasticsearch_collectd.py:531]: :+1: okay fair

File: elasticsearch_collectd.py (hunk 3 above)
Comment (gtschurwald) [reply elasticsearch_collectd.py:664]: No that should be the same, good catch.

**Comment by gtschurwald** [comment]
Merging this, has been running succesfully on qa-ea-0 nodes for a week now.

**Pull request merged**